mm/kfence/core.c

```diff
@@ -217,6 +223,28 @@ static inline void for_each_canary(const struct kfence_metadata *meta, bool (*fn
 	}
 }
 
+/* Return if the allocation can be serviced by KFENCE. */
+static bool kfence_can_alloc(struct kmem_cache *cache, size_t size)
+{
+	int pool_usage;
+
+	if (size > PAGE_SIZE)
+		return false;
+
+	pool_usage = (100 * atomic_long_read(&counters[KFENCE_COUNTER_ALLOCATED])) /
+		     CONFIG_KFENCE_NUM_OBJECTS;
+	if (pool_usage > min(kfence_non_kmalloc_cutoff, 100U)) {
+		/*
+		 * Iterating through all kmem_caches is more expensive; just
+		 * checking the prefix of the name is simpler.
+		 */
+		if (!str_has_prefix(cache->name, "kmalloc"))
```

Comment:
Note that there are "dma-kmalloc-XXX" caches as well.

Reply:
If we've exhausted 80% of our pool, I'd say we're fine just servicing normal kmalloc(), and can safely ignore DMA things.